File: digitalRoot.js
function digital_root(n) {
  let arr = n.toString().split('');
  let numArr = arr.map(Number);
  let newNum = numArr.reduce((sum, num) => sum + num);
  if (newNum.toString().length <= 1) {
    return newNum;
  } else {
    return digital_root(newNum);
  }
}

function digital_root(n) {
  return (n - 1) % 9 + 1;
}

function digital_root(n) {
  if (n < 10) return n;
  return digital_root(n % 10 + digital_root(Math.floor(n / 10)));
}

digital_root(16);

File: digital_root.rb
def digital_root(n)
  return 0 if n.zero?
  1 + (n - 1) % 9
end

def digital_root(n)
  n < 10 ? n : digital_root(n / 10 + n % 10)
end

def digital_root(n)
 n < 10 ? n : digital_root(n.digits.sum)
end

digital_root(16);
